demo/src/screens/MainScreen.js

```diff
@@ -33,21 +34,32 @@ class MainScreen extends Component {
     };
   }
 
+  settingsScreenName = 'unicorn.Settings';
+
   constructor(props) {
     super(props);
 
     const data = props.navigationData || navigationData;
+    const extraSettingsUI = props.extraSettingsUI;
+    const showRefreshAppMessage = props.showRefreshAppMessage;
 
     this.state = {
       currentPage: 0,
-      filteredNavigationData: data
+      filteredNavigationData: data,
+      extraSettingsUI,
+      showRefreshAppMessage
     };
 
     this.filterExplorerScreens = _.throttle(this.filterExplorerScreens, 300);
 
     Navigation.events().bindComponent(this);
   }
 
+  shouldComponentUpdate = (nextProps, nextState) => {  
+    this.updateShowRefreshAppMessage(nextProps.showRefreshAppMessage);
+    return true;
+  }
+
```

Comment:
This will be called before each render. 
Not sure that's the right way to do that.. 

`shouldComponentUpdate` is meant to have calculation regarding whether the component should render or not. 
If anything you probably want to move this part of code to `componentDidUpdate` .

Anyway, maybe there's a better way to trigger a toast from the private repo.. 
I think the wiring here is a little more complicated then is should be.

Reply:
`componentDidUpdate` - gotcha. 

@ethanshar  Triggering the toast from the private repo - you mean that the toast should be rendered in the private repo?
or some different - easier wiring?

Reply:
Yes, something like that. 
In the private repo we have an infrastructure for trigger Toast from anywhere in the code, we call it global toast. 
It's a functionality we offer to One App and we use it ourselves in the demo app. so it might easier for you to use that.. 
Search in the private repo for "toggleToast" in the MainScreen.